generators/upgrade/index.js

```diff
@@ -211,34 +209,58 @@ module.exports = class extends BaseGenerator {
-                        );
-                    }
+                Promise.all(
+                    this.blueprints
+                        .filter(blueprint => {
+                            if (this.targetBlueprintVersions && this.targetBlueprintVersions.length > 0) {
+                                const targetBlueprint = this.targetBlueprintVersions.find(elem => {
+                                    return elem.name === blueprint.name;
+                                });
+                                if (targetBlueprint && targetBlueprint.version && targetBlueprint.version !== 'latest') {
+                                    this.log(
+                                        `Blueprint ${targetBlueprint.name} will be upgraded to target version: ${targetBlueprint.version}`
+                                    );
+                                    blueprint.latestBlueprintVersion = targetBlueprint.version;
+                                    return false;
+                                }
+                            }
+                            return true;
+                        })
+                        .map(blueprint => {
+                            return new Promise(resolve => {
+                                this._retrieveLatestVersion(blueprint.name, latestVersion => {
+                                    blueprint.latestBlueprintVersion = latestVersion;
+                                    if (semver.lt(blueprint.version, blueprint.latestBlueprintVersion)) {
+                                        this.newBlueprintVersionFound = true;
+                                        this.success(`New ${blueprint.name} version found: ${this.latestBlueprintVersion}`);
+                                    } else if (this.force) {
+                                        this.newBlueprintVersionFound = true;
+                                        this.log(chalk.yellow('Forced re-generation'));
+                                    } else {
+                                        if (this.newBlueprintVersionFound === undefined) {
+                                            this.newBlueprintVersionFound = false;
+                                        }
+                                        this.warning(
+                                            `${chalk.green(
+                                                'No update available.'
+                                            )} Application has already been generated with latest version for blueprint: ${blueprint.name}`
+                                        );
+                                    }
+                                    this.success(`Done checking for new version for blueprint ${blueprint.name}`);
+                                    resolve();
+                                });
+                            });
+                        })
+                ).then(() => {
+                    this.success('Done checking for new blueprint versions');
```

Comment:
`new version of blueprints`? (beware I'm not a native English speaking person 😉)

Reply:
Not an native English speaker either :) Fixed